source/extensions/clusters/original_dst/original_dst_cluster.cc

```diff
@@ -105,7 +105,7 @@ HostConstSharedPtr OriginalDstCluster::LoadBalancer::chooseHost(LoadBalancerCont
 Network::Address::InstanceConstSharedPtr
 OriginalDstCluster::LoadBalancer::filterStateOverrideHost(LoadBalancerContext* context) {
   const auto streamInfos = {
-      context->requestStreamInfo(),
+      static_cast<const StreamInfo::StreamInfo*>(context->requestStreamInfo()),
```

Comment:
I think we should use const_cast for this case.

Reply:
Done